dask_geomodeling/raster/sources.py

```diff
@@ -207,87 +317,57 @@ def get_sources_and_requests(self, **request):
         )
         if start is None:
             return [({"mode": "empty_" + request["mode"]}, None)]
+        bands = band1, band2 + 1
+
+        # create a dataset
+        array = self.data
+        kwargs = {
+            "no_data_value": float(self.no_data_value),
+            "geo_transform": self.geo_transform,
+            "projection": osr.GetUserInputAsWKT(self.projection),
+        }
+
+        driver = gdal.GetDriverByName("mem")
+        with utils.Dataset(array, **kwargs) as _dataset:
+            dataset = driver.CreateCopy("", _dataset)
+        if self.metadata is not None:
+            for i in range(len(self.metadata)):
+                band = dataset.GetRasterBand(i + 1)
+                band.SetMetadataItem("metadata", self.metadata[i])
 
         # set the process kwargs depending on the mode
         if mode == "vals":
             process_kwargs = {
                 "mode": "vals",
-                "data": self.data[first_i : last_i + 1],
-                "no_data_value": self.no_data_value,
+                "dataset": dataset,
```

Comment:
I am a bit worried whether this will survive the `dask` process of submitting tasks to external compute services. For a singleprocess compute mode this will be fine, but multiprocessing or compute clusters will probably pose a problem because a gdal dataset can't be serialized by dask.

As we are not using this in production, I guess it is OK to do it like this (and document this), but as I see it it is probably easy to serialize the things that are needed to construct the dataset instead?

Reply:
Good point, refactored it.